pycalphad/core/calculate.py

```diff
@@ -316,7 +316,7 @@ def calculate(dbf, comps, phases, mode=None, output='GM', fake_points=False, bro
     largest_energy = 1e10
 
     # Consider only the active phases
-    list_of_possible_phases = filter_phases(dbf, comps)
+    list_of_possible_phases = filter_phases(dbf, comps, phases)
     active_phases = sorted(set(list_of_possible_phases).intersection(set(phases)))
```

Comment:
The new changes permit check evaluate the intersection between `candidate_phases` and database phases within `filter_phases`. So, what do you think about move `active_phases = sorted(set(list_of_possible_phases).intersection(set(phases)))` to `filter_phases`?

Reply:
Yes, I think that would now be correct behavior.